pkg/broker/handler/processors/deliver/processor_test.go

```diff
@@ -318,12 +325,107 @@ func (NoReplyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 }
 
 func BenchmarkDeliveryNoReply(b *testing.B) {
-	sampleEvent := newSampleEvent()
 	httpClient := http.Client{
 		Transport: &http.Transport{
 			MaxIdleConnsPerHost: runtime.NumCPU(),
 		},
 	}
+	targetSvr := httptest.NewServer(NoReplyHandler(struct{}{}))
+	defer targetSvr.Close()
+
+	benchmarkNoReply(b, httpClient, targetSvr.URL)
+}
+
+type ReplyHandler struct {
+	msg binding.Message
+}
+
+func (h ReplyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	cehttp.WriteResponseWriter(req.Context(), h.msg, http.StatusOK, w)
+}
+
+func BenchmarkDeliveryWithReply(b *testing.B) {
+	httpClient := http.Client{Transport: &http.Transport{MaxIdleConnsPerHost: runtime.NumCPU()}}
+	ingressSvr := httptest.NewServer(NoReplyHandler(struct{}{}))
+	defer ingressSvr.Close()
+	benchmarkWithReply(b, ingressSvr.URL, func(b *testing.B, reply *event.Event) (http.Client, string) {
+		targetSvr := httptest.NewServer(ReplyHandler{msg: binding.ToMessage(reply)})
+		b.Cleanup(targetSvr.Close)
+		return httpClient, targetSvr.URL
+	})
+}
+
+type fakeRoundTripper map[string]*http.Response
+
+func (r fakeRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if req.Body != nil {
+		defer req.Body.Close()
+	}
+	if resp, ok := r[req.URL.String()]; ok {
+		return resp, nil
+	}
+	return nil, fmt.Errorf("URL not found: %q", req.URL)
+}
+
+func BenchmarkDeliveryNoReplyFakeClient(b *testing.B) {
+	targetAddress := "target"
+	httpClient := http.Client{
+		Transport: fakeRoundTripper(map[string]*http.Response{
+			targetAddress: {
+				StatusCode: 202,
+				Header:     make(map[string][]string),
+				Body:       fakeBody{new(bytes.Buffer)},
+			},
+		}),
+	}
+	benchmarkNoReply(b, httpClient, targetAddress)
+}
+
+type fakeBody struct {
+	*bytes.Buffer
+}
+
+func (b fakeBody) Close() error {
+	return nil
+}
+
+func makeFakeTargetWithReply(b *testing.B, reply *event.Event) (httpClient http.Client, targetAddress string) {
+	req, err := http.NewRequest("POST", "", nil)
+	if err != nil {
+		b.Fatal(err)
+	}
+	cehttp.WriteRequest(context.Background(), binding.ToMessage(reply), req)
+	body := new(bytes.Buffer)
+	if req.Body != nil {
+		io.Copy(body, req.Body)
+		req.Body.Close()
+	}
+
+	httpClient = http.Client{
+		Transport: fakeRoundTripper(map[string]*http.Response{
+			fakeTargetAddress: {
+				StatusCode: 200,
+				Header:     req.Header,
+				Body:       fakeBody{body},
+			},
+			fakeIngressAddress: {
+				StatusCode: 202,
+				Header:     make(map[string][]string),
+				Body:       fakeBody{new(bytes.Buffer)},
+			},
+		}),
+	}
+	targetAddress = fakeTargetAddress
+	return
+}
+
+func BenchmarkDeliveryWithReplyFakeClient(b *testing.B) {
+	benchmarkWithReply(b, fakeIngressAddress, makeFakeTargetWithReply)
+}
```

Comment:
Add a comment similar to what you've put in the PR description. This helps us later know that the one with the fake client could be a baseline.